src/components/Personalization/createViewChangeHandler.js

```diff
@@ -10,17 +10,34 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-export default ({ executeCachedViewDecisions, viewCache, showContainers }) => {
+import { NOT_RENDERED } from "./constants/decisionStatus";
+
+export default ({
+  executeCachedViewDecisions,
+  viewCache,
+  addRenderedDecisionStatus,
+  formatDecisions,
+  showContainers
+}) => {
   return ({ personalizationDetails, onResponse, onRequestFailure }) => {
     const viewName = personalizationDetails.getViewName();
-
+    const executedDecisionsResult = [];
     if (personalizationDetails.isRenderDecisions()) {
-      executeCachedViewDecisions({ viewName });
-      return;
+      executeCachedViewDecisions({ viewName }).then(result => {
+        executedDecisionsResult.push(result);
+      });
```

Comment:
I don't really know what I'm talking about here, but should we be waiting for this promise? It seems like there might be a race condition where our `onResponse` callback reports `executedDecisionsResult` before `executedDecisionsResult.push` has had a chance to be called.